src/lightning_app/core/constants.py

```diff
@@ -13,7 +14,12 @@
 FLOW_DURATION_SAMPLES = 5
 
 APP_SERVER_HOST = os.getenv("LIGHTNING_APP_STATE_URL", "http://127.0.0.1")
-APP_SERVER_PORT = 7501
+APP_SERVER_PORT = int(
+    os.getenv("LIGHTNING_APP_SERVER_PORT", 7501 if is_free_network_port(7501) else find_free_network_port())
+)
+os.environ["LIGHTNING_APP_SERVER_PORT"] = str(
+    APP_SERVER_PORT
+)  # We set this to allow child processes (like streamlit server) to know where the app_server is actually is.
```

Comment:
Now that this is not a constant anymore, this call should move directly to where it is used. 
I highly recommend not to call `find_free_network_port()` here, because this will run at import time (and thus adds to import time).

I would leave this constant here and rename it to `APP_SERVER_DEFAULT_PORT`, then choose the appropriate port at the right place.